src/app/api/_utils/slack.ts

```diff
@@ -24,22 +24,31 @@ export function getSlackMessage(responseType: SlackResponseType, responseText: s
 }
 
 async function fetchSlackApi(slackApiString: string, team_id: string) {
-  // fetch the organization:
+  // Fetch the organization:
   const organization = await PostgresClient.getOrganization(team_id);
   const token = decrypt(organization.access_hash);
 
   // const res = await fetch('https://slack.com/api/usergroups.users.list?usergroup=' + userGroupId,
-  const res = await fetch('https://slack.com/api/' + slackApiString,
-    {
-      method: 'GET',
-      headers: {
-        'Authorization': 'Bearer ' + token, // TODO: we need to take the token we stored upon installation, not this one.
-      },
-    }
-  );
-
-  // if this fails, we must throw an error here, and handle it in the routes. something like: "the token is expired, please reinstall the app"
-  return await res.json();
+  const res = await fetch('https://slack.com/api/' + slackApiString, {
+    method: 'GET',
+    headers: {
+      'Authorization': 'Bearer ' + token,
+    },
+  });
+
+  const jsonResponse = await res.json();
+
+  if (jsonResponse.error === 'invalid_auth') {
```

Comment:
this is going to fail
```suggestion
  if (jsonResponse?.error === 'invalid_auth') {
```